### ISSUE TITLE
[TTT] Apply changes from Llama generator to TTT generator

### ISSUE THREAD

**Issue by djordje-tt** [tenstorrent/tt-metal#35731]
### Bug description

Once Llama70b modifies SamplingGenerator that's reflected to TTT due to code sharing. Being that said, even though Llama3.1-8b is not affected by non-uniform seeding failure it also needs adaptation to behave in the similar fasion.

**Comment by djordje-tt** [comment]
On `divanovic/seed-fix` I applied Llama3.3-70b changes to TTT Llama3.1-8b. I observed different outputs across users so I fixed seed in ttnn.sampling. 

In iteration 47, users 5 and 9 have different logits than other 30 users, here is snapshot below. I am debugging to figure out which op is causing this. Below is snapshot for logits. This is not going to make it for today!

<img width="972" height="867" alt="Image" src="https://github.com/user-attachments/assets/a894a1b5-6658-4881-9402-47245064fa94" />

@bgoelTT @ekuslysTT

**Comment by bgoelTT** [comment]
Thanks @djordje-tt, curiously, because this test was passing before what happens if you don't apply the Llama-70B op changes? Or are the changes required to make the seed reseting work?

**Comment by djordje-tt** [comment]
So we are using same module models/common/sampling so we need to make it work on TTT. Otherwise, we would need manage 2 different implementation which would introduce another degree of complexity for future updates/maintenance/development.

**Comment by bgoelTT** [comment]
Thank you for the explanation, understood

**Comment by djordje-tt** [comment]
Further debugging showed that without penalties we are seeing same outputs for all the users. I tracked bug further to see how penalties change output tokens and it seems that counts start to differ very early but affect the final token outputs just at iteration 47. Penalties update stats does not work properly and I am debugging it at the moment. Once that is fixed we should be good to close the issue. This seems like the only source of divergence and I hope to get to the bottom of the bug today and run CI pipelines tomorrow morning. 

@ekuslysTT
Being that said, I think we should have the issue fixed and tested tomorrow!

**Comment by djordje-tt** [comment]
- Bug is fixed, so between prefill and decode there was no resetting penalties stats so that was source of divergence that affected later tokens to be different across users. 
- ✅ Llama3.1-8b Models CI: https://github.com/tenstorrent/tt-shield/actions/runs/21030872736
- ✅ Llama3.3-70b Models CI: https://github.com/tenstorrent/tt-shield/actions/runs/21034239371
